### DISCUSSION TITLE
How can I digitally sign MSI installer

### DISCUSSION THREAD

**Discussion by HarshSuman82** [wixtoolset/issues#6488] [Questions]
I want to create an MSI installer file as digitally signed. I tried to put the event comment (CD "C:\Program Files (x86)\Windows Kits\10\bin\x64" signtool.exe sign /f "[Path Of pfx file]" /p "password" /t "[Path of MSI file]") at WiX toolset project's Post Build Event.
But I am getting an error **"exited with code 9009"**. Please help me to find the solution for this.

**Comment by BMurri** [comment]
It's recommended to use `<Target Name="SignMsi">` in your wixproj file as described [here](https://wixtoolset.org/documentation/manual/v3/overview/insignia.html).

**Reply by HarshSuman82** [reply]
http://wixtoolset.org/documentation/manual/v3/overview/insignia.html

This Insignia talks about external cabs but in my case, I don't have an external cab.  It also doesn't mention a pfx file.  I have to mention pfx file for signing the MSI file

**Reply by BMurri** [reply]
If you don't have external cabs you don't need to include the `SignCab` target.

In that page, the content of the targets are just examples. Modify them as appropriate to your needs.
